src/utils/mtev_hash.h

```diff
@@ -38,11 +38,25 @@
 
 #include "mtev_config.h"
 #include <ck_hs.h>
+#include <ck_spinlock.h>
 
 typedef void (*NoitHashFreeFunc)(void *);
 
-typedef struct {
+typedef enum mtev_hash_lock_mode {
+  MTEV_HASH_LOCK_MODE_NONE = 0,
+  MTEV_HASH_LOCK_MODE_MUTEX = 1,
+  MTEV_HASH_LOCK_MODE_SPIN = 2,
+  MTEV_HASH_LOCK_MODE_SIZE
```

Comment:
why add MTEV_HASH_LOCK_MODE_SIZE? on some compilers, it'll generate a warning in the `switch` in `set_lock_mode_funcs` that some enum values aren't handled.